lib/Importer/ONNXModelLoader.cpp

```diff
@@ -256,6 +256,96 @@ bool ONNXModelLoader::loadOperator(const ONNX_NAMESPACE::NodeProto &op) {
     return true;
   }
 
+  if (typeName == "Slice") {
+    auto data = getNodeValueOrCreateConstantByName(op.input(0));
+    auto dims = data.dims();
+    auto numDims = dims.size();
+
+    // Attributes 'starts' and 'ends' are mandatory and must be consistent.
+    assert(dict.count("starts") && "Slice: attribute 'starts' is mandatory.");
+    assert(dict.count("ends") && "Slice: attribute 'ends' is mandatory.");
+    auto starts = getShape<ssize_t>(dict["starts"]);
+    auto ends = getShape<ssize_t>(dict["ends"]);
+    assert((starts.size() == ends.size()) &&
+           "Slice: 'starts' and 'end' arrays must have the same size.");
+
+    // Attribute 'axes' is optional.
+    std::vector<ssize_t> axes;
+    if (dict.count("axes")) {
+      // The ONNX spec is unclear so we consider that the 'axes' array may have
+      // any size. The constraints are:
+      // - the element value must be in range [0, numDims[,
```

Comment:
closing ], not [

Reply:
`[` means that  `numDims` is excluded from the range. Do you have another notation in the US ?

Reply:
`[0, numDims)`

Reply:
@jfix71 thanks ! I changed the notation.

Reply:
Ha, wow, sorry, I thought it was just a typo.  That's my i18n lesson for the day!